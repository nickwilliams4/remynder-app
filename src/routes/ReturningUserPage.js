import React, { Component } from 'react'
import ReturningUser from '../ReturningUser/ReturningUser'
import { Section } from '../Utils/Utils'

export default class LoginPage extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/NotePage'
    history.push(destination)
  }

  render() {
    return (
      <Section className='LoginPage'>
        <ReturningUser
          onLoginSuccess={this.handleLoginSuccess}
        />
      </Section>
    )
  }
}