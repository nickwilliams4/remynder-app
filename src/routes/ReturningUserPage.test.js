import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReturningUserPage from './ReturningUserPage'

describe(`ReturningUserPage component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
  }

  it('renders the ReturningUser', () => {
    const wrapper = shallow(<ReturningUserPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})