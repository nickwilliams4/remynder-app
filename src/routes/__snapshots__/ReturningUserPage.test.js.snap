// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReturningUserPage component renders the ReturningUser 1`] = `
<Section
  className="LoginPage"
>
  <ReturningUser
    onLoginSuccess={[Function]}
  />
</Section>
`;
